### PR TITLE
#2737 converted RecentSearchDao.java to RecentSearchDao.kt

### LINKED ISSUE #2737: convert RecentSearchDao.java to Kotlin
Parent issue: #1826

What changes you will do:
will convert Java file to Kotlin

I couldn't find an issue regarding this, do tell me if there is I will close this

### PR DESCRIPTION
**Pull request by cdhiraj40**
Fixes #2737

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        while (searchCursor.moveToNext()) {
+          result.add(RecentSearch(searchCursor))
+        }
+      }
+      return result
+    }
+
+  companion object {
+    private const val NUM_RECENT_RESULTS = 5
+  }
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt:49]: We can get rid of this.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+@Deprecated("")
+class RecentSearchDao @Inject constructor(private val mDb: KiwixDatabase) {
+  /**
+   * Returns a distinct enumeration of the `NUM_RECENT_RESULTS` most recent searches.
+   */
+  val recentSearches: List<RecentSearch>
+    get() {
+      val result: MutableList<RecentSearch> = ArrayList()
+      mDb.query(
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt:37]: I think we should make this a function

Comment (cdhiraj40) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt:37]: any preferable function name?

Comment (cdhiraj40) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt:37]: i will go with ``getRecentSearches``

**Comment by cdhiraj40** [comment]
please review @gouri-panda

**Comment by cdhiraj40** [comment]
please review @gouri-panda

**Comment by kelson42** [comment]
@gouri-panda Can you please make a new review pass on this PR?

**Comment by cdhiraj40** [comment]
@gouri-panda @kelson42 My exams are going on, but soon I will update both the PRs. This, #2764 and #2747. Hope you understand.

**Comment by kelson42** [comment]
@cdhiraj40 Woukd you be able please to update this PR so we can review and merge within 3.6.0?

**Comment by cdhiraj40** [comment]
> @cdhiraj40 Woukd you be able please to update this PR so we can review and merge within 3.6.0?

hey sorry for late reply, will update later tonight itself.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  fun getRecentSearches(): MutableList<RecentSearch> {
+    val result: MutableList<RecentSearch> = ArrayList()
+    mDb.query(
+      RecentSearch::class.java, Query.select()
+    ).use { searchCursor ->
+      while (searchCursor.moveToNext()) {
+        result.add(RecentSearch(searchCursor))
+      }
+    }
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/local/dao/RecentSearchDao.kt:39]: @cdhiraj40 We should catch with an exception.

**Comment by cdhiraj40** [comment]
okay! @kelson42 @gouri-panda will do the changes soon!
could you also look at it here and let me know what has to be done here: https://github.com/kiwix/kiwix-android/pull/2747#issuecomment-1175211361 ?

**Comment by cdhiraj40** [comment]
@gouri-panda I think this is done? could you review it once? Thanks

**Comment by kelson42** [comment]
git rebase devwlop. But forst you shozkd remove the merge commit. You should not squash. Google is your friend.

**Comment by gouri-panda** [comment]
For some reason, the CI keeps failing :(

**Comment by kelson42** [comment]
@cdhiraj40 What’s up here, we need to finish this PR ASAP to be able to release 3.6.0

**Comment by cdhiraj40** [comment]
> @cdhiraj40 What’s up here, we need to finish this PR ASAP to be able to release 3.6.0

well I am done from my side, I have no idea why the CI keeps failing 🤔

**Comment by kelson42** [comment]
@MohitMaliFtechiz Can you please fix and merge?

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz Can you please fix and merge?

sure

**Review by MohitMaliFtechiz — APPROVED** [review]
changes are approved.

**Pull request merged**